src/structure_managers/property_block_sparse.hh

```diff
@@ -444,92 +453,134 @@ namespace rascal {
+
+    template <size_t CallerOrder, size_t CallerLayer>
+    inline decltype(auto)
+    operator()(const ClusterRefKey<CallerOrder, CallerLayer> & id,
+               const Key_t & key) {
+      static_assert(CallerOrder <= Order, "should be CallerOrder <= Order");
+      static_assert(CallerLayer >= PropertyLayer,
+                    "You are trying to access a property that does not exist at"
+                    "this depth in the adaptor stack.");
+
+      return this->operator()(id.get_cluster_index(CallerLayer),
+                              CallerOrder - 1, key);
     }
 
-    inline size_t get_dense_feature_size(const size_t & index) {
-      auto keys = this->values[index].get_keys();
-      return this->get_nb_comp() * keys.size();
+    //! Accessor for property by index for dynamically sized properties
+    inline DenseRef_t operator()(const size_t & index, const size_t & i_order,
+                                 const Key_t & key) {
+      return DenseRef_t(&this->values[i_order][index].at(key)(0, 0),
+                        this->values[i_order][index].at(key).rows(),
+                        this->values[i_order][index].at(key).cols());
     }
 
+    // inline size_t get_dense_feature_size(const size_t & index) {
+    //   auto keys{this->values[0][index].get_keys()};
+    //   return this->get_nb_comp() * keys.size();
+    // }
+
     //! Accessor for property by cluster index and return a dense
     //! representation of the property associated to this cluster
-    template <size_t CallerLayer>
-    inline Dense_t get_dense_row(const ClusterRefKey<Order, CallerLayer> & id) {
+    template <size_t CallerOrder, size_t CallerLayer>
+    inline Dense_t
+    get_dense_row(const ClusterRefKey<CallerOrder, CallerLayer> & id) {
+      static_assert(CallerOrder <= Order, "should be CallerOrder <= Order");
       static_assert(CallerLayer >= PropertyLayer,
                     "You are trying to access a property that does not exist at"
                     "this depth in the adaptor stack.");
 
-      return this->get_dense_row(id.get_cluster_index(CallerLayer));
+      return this->get_dense_row(id.get_cluster_index(CallerLayer),
+                                 CallerOrder - 1);
     }
 
-    inline Dense_t get_dense_row(const size_t & index) {
-      auto keys = this->values[index].get_keys();
+    inline Dense_t get_dense_row(const size_t & index, const size_t & i_order) {
+      auto keys = this->values[i_order][index].get_keys();
       Dense_t feauture_row = Dense_t::Zero(this->get_nb_comp(), keys.size());
       size_t i_col{0};
       for (const auto & key : keys) {
         size_t i_row{0};
-        for (int i_pos{0}; i_pos < this->values[index][key].size(); i_pos++) {
-          feauture_row(i_row, i_col) = this->values[index][key](i_pos);
+        for (int i_pos{0}; i_pos < this->values[i_order][index][key].size();
+             i_pos++) {
+          feauture_row(i_row, i_col) = this->values[i_order][index][key](i_pos);
```

Comment:
s/feauture/feature ? (Unless it's supposed to be pronounced with a French accent?)
```suggestion
          feature_row(i_row, i_col) = this->values[i_order][index][key](i_pos);
```